projects/plugins/social/src/class-jetpack-social.php

```diff
@@ -242,6 +244,8 @@ public function activate_module_on_plugin_activation() {
 		if ( get_option( self::JETPACK_SOCIAL_ACTIVATION_OPTION ) && $this->is_connected() ) {
 			delete_option( self::JETPACK_SOCIAL_ACTIVATION_OPTION );
 			( new Modules() )->activate( self::JETPACK_PUBLICIZE_MODULE_SLUG, false, false );
+			wp_safe_redirect( esc_url( admin_url( 'admin.php?page=' . JETPACK_SOCIAL_PLUGIN_SLUG ) ) );
+			exit;
```

Comment:
Do we need the `exit` here?

Reply:
Yes, I think it's because the redirect can be made conditional. In the docs it says:

> Note: [wp_safe_redirect()](https://developer.wordpress.org/reference/functions/wp_safe_redirect/) does not exit automatically, and should almost always be followed by a call to exit;:

Reply:
Looks like this will only work if the site was connected before? We should also check that we're only activating the plugin from the plugins page before attempting the redirection.

Reply:
Yes, that's partly way we held off on this at the time. It seemed like redirecting all the time could be a bit risky. What if the plugin is activated in 'My Jetpack' or cycled during an upgrade.

I think Backup or one of the other plugins was doing something similar, so we can check if there is an approach we could standardise on.

> Looks like this will only work if the site was connected before?

Yes good point! It shouldn't matter if we have a connection or not.